File: Java/src/HashTable/HashTableADT.java
package HashTable;

public interface HashTableADT<K, V> extends Iterable<K> {
    int size();

    boolean isEmpty();

    int hashCodeToIndex(int hashedKey);

    void clear();

    boolean has(K key);

    V insert(K key);

    V get(K key);

    V remove(K key);
}

File: Typescript/src/HashTable/HashTableADT.ts
export interface HashTableADT<K extends string | number, V>
  extends Iterable<K> {
  size(): number;

  isEmpty(): boolean;

  hashCodeToIndex(hashedKey: number): number;

  clear(): void;

  has(key: K): boolean;

  insert(key: K, value: V): V;

  get(key: K): V | undefined;

  remove(key: K): V | undefined;
}
